### PR TITLE
[PLAT-2139] Ensure session counts are reported thread safe

### PR DESCRIPTION
**Pull request by Pezzah**
## Goal

Currently if there are multiple errors in quick succession, then the number of errors reported in the current session may be incorrect.

This will clone the current session when the metrics are updated to ensure that this cannot happen

## Review

For the submitter, initial self-review:

- [ ] Commented on code changes inline explain the reasoning behind the approach
- [ ] Reviewed the test cases added for completeness and possible points for discussion
- [ ] A changelog entry was added for the goal of this pull request
- [ ] Check the scope of the changeset - is everything in the diff required for the pull request?
- This pull request is ready for:
  - [ ] Initial review of the intended approach, not yet feature complete
  - [ ] Structural review of the classes, functions, and properties modified
  - [ ] Final review

For the pull request reviewer(s), this changeset has been reviewed for:

- [ ] Consistency across platforms for structures or concepts added or modified
- [ ] Consistency between the changeset and the goal stated above
- [ ] Internal consistency with the rest of the library - is there any overlap between existing interfaces and any which have been added?
- [ ] Usage friction - is the proposed change in usage cumbersome or complicated?
- [ ] Performance and complexity - are there any cases of unexpected O(n^3) when iterating, recursing, flat mapping, etc?
- [ ] Concurrency concerns - if components are accessed asynchronously, what issues will arise
- [ ] Thoroughness of added tests and any missing edge cases
- [ ] Idiomatic use of the language

### CODE REVIEW

File: bugsnag/src/main/java/com/bugsnag/Report.java [hunk 1]
```diff
@@ -131,23 +131,22 @@ public String getContext() {
 
     @Expose
     Map<String, Object> getSession() {
+        return sessionMap;
+    }
+
+    void setSession(Session session) {
         if (session == null) {
-            return null;
+            sessionMap = null;
```
Comment (fractalwrench) [inline bugsnag/src/main/java/com/bugsnag/Report.java]: We should add a unit test to verify that if the session parameter is null, the payload doesn't contain any session information. The current implementation will cause a crash due to NPEs.
